public/event-image-upload.js

```diff
@@ -0,0 +1,40 @@
+(function() {
+  var fileInput = document.getElementById('eventFileInput');
+  var fileErrorMessage = document.getElementById('fileErrorMessage');
+  var hiddenFileInput = document.getElementById('hiddenFileInput');
+  var spinner = document.getElementById('spinner');
+  var imagePreview = document.getElementById('imagePreview');
+
+  fileInput.onchange = function() {
+    var fileInputFiles = fileInput.files;
+    var file = fileInputFiles[0];
+    imagePreview.src && (imagePreview.src = '');
```

Comment:
Do not put side effects in a logical expression

correct
```js
x = y && y.x
```
incorrect
```js
y && (x = y.x)
```
correct
```js
if (y) x = y.x
```

Inside logical expression / ternary you should not put side effects